File: algorithms/2_medium/L/01503_last_moment_before_all_ants_fall_out_of_a_plank/solution.c
int getLastMoment(const int n, const int *left, const int left_len, const int *right, const int right_len) {
    int res = 0;
    for (int i = 0; i < left_len; i += 1) {
        res = MAX(res, left[i]);
    }
    for (int i = 0; i < right_len; i += 1) {
        res = MAX(res, n - right[i]);
    }
    return res;
}

File: algorithms/2_medium/L/01503_last_moment_before_all_ants_fall_out_of_a_plank/solution.rs
impl Solution {
    pub fn get_last_moment(n: i32, left: Vec<i32>, right: Vec<i32>) -> i32 {
        let mut res = 0;
        for l in left {
            res =res.max(l);
        }
        for r in right {
            res = res.max(n - r);
        }
        res
    }
}
